### PR TITLE
add GET /api/articles/:article_id/comments endpoint

### PR DESCRIPTION
**Pull request by dheat23**
add GET /api/articles/:article_id/comments endpoint with testing for 200, 400 & 404 responses

### CODE REVIEW

**Comment by dheat23** [comment]
going to refactor the 404 error/add in a 200 for article id exists but has no comments

File: __tests__/app.test.js [hunk 1]
```diff
@@ -124,4 +124,51 @@ describe('GET /api/articles', () => {
...
   });
+});
+describe('GET /api/articles/:article_id/comments', () => {
+    test('200: responds with array of comments for given article id with most recent comments first', () => {
+        return request(app)
+        .get("/api/articles/1/comments")
+        .expect(200)
+        .then(({body}) => {
+            const {comments} = body;
+            expect(comments).toHaveLength(11);
```
Comment (2202Hannah) [inline __tests__/app.test.js:135]: Great job remembering to check that the array is not empty when asserting inside a forEach

File: app/models.js [hunk 2]
```diff
@@ -23,4 +23,21 @@ exports.selectAllArticles = () => {
...
+};
+
+exports.selectCommentsByArticleId = (id) => {
+    return db.query(`SELECT comments.comment_id, comments.votes, comments.created_at, comments.author, comments.body, comments.article_id FROM comments JOIN articles ON comments.article_id = articles.article_id WHERE articles.article_id = $1 ORDER BY comments.created_at DESC;`, [id])
+    .then(({rows}) => {
+            return rows
+    })
+};
+
+exports.checkArticleExists = (value) => {
```
Comment (2202Hannah) [inline app/models.js:35]: Nicely done!

**Review by 2202Hannah — APPROVED** [review]
Really nice work on this ticket. You have considered the distinction between an empty array being returned because there are no comments for an ID that exists and an empty array being returned because the article_id does not exist and handled this well. 

I like that you have used Promise.all() in your controller when checking the article exists.

You can merge this to main when you are ready.

**Pull request merged**